lib/floe/workflow/path.rb

```diff
@@ -34,7 +34,18 @@ def value(context, input = {})
         return obj if path == "$"
 
         results = JsonPath.on(obj, path)
-        results.count < 2 ? results.first : results
+        case results.count
+        when 0
+          raise Floe::PathError, "references an invalid value"
```

Comment:
@kbrock does this match what StepFunctions does?  Or would this just return `null`?

Reply:
Both Intrinsics and Choice threw errors if the path is not present (except for IsPresence checks).
That is where I found the issue with `IsPresent` vs `IsNull` - present checks that it is defined vs null checks that it is defined, but null.

Now datatype comparison was inconsistent, but presence was consistent.